pkg/ecdsa/tss/tss.go

```diff
@@ -169,3 +169,139 @@ func (s *ThresholdSigner) CalculateSignature(
 
 	return signature, err
 }
+
+func BroadcastRecoveryAddress(
+	parentCtx context.Context,
+	groupID string,
+	memberID MemberID,
+	groupMemberIDs []MemberID,
+	dishonestThreshold uint,
```

Comment:
This def feels squirmy to me. I wonder if we should be instantiating an extension type of `ThresholdSigner` that embeds `ThresholdSigner`, that is specific to the application in question. In general I think we need to make applications more first-class citizens rather than concepts that are strapped to the side of the system (you can see a lot of that at play in the refactorings in #693).

Reply:
That's a very good point, the additional logic is application-specific. 👌

Reply:
Flagging that this is part of the refactoring we're punting as per the note [above](https://github.com/keep-network/keep-ecdsa/pull/691#discussion_r577748928).